### ISSUE TITLE
ForbiddenComment should allow patterns

### ISSUE THREAD

**Issue by TWiStErRob** [detekt/detekt#3508]
Following up on https://github.com/detekt/detekt/issues/959, the resolution of that is not powerful enough / complicates things.

## Context
We have the following configuration:
```yaml
  ForbiddenComment:
    values: ['REVIEW', 'STOPSHIP']
```
and I got this message flagged by Detekt:
```kotlin
// to express in the preview that it's not a normal TextView.
```
the only workaround I found is this
```yaml
  ForbiddenComment:
    values: ['REVIEW', 'STOPSHIP']
    # Allow "preview", "FireView", "DoReView" to occur in comments.
    # So only `// REVIEW explanation` will be flagged.
    # See https://github.com/detekt/detekt/issues/3508
    allowedPatterns: '(?<!^)(review|Review|reView|ReView)'
```

## Expected Behavior of the rule
```kotlin
// REVIEW foo -> flag
// review foo -> flag
// foo review -> allow
// preview -> allow
```
To do this I want to be able to say [`^(?i)REVIEW\b`](https://regex101.com/r/NioVGu/3) (as in case insensitive, starts with the letters "review" and it's not followed by word characters), instead of the current behavior which is https://regex101.com/r/NioVGu/2/

**Comment by TWiStErRob** [comment]
Even this workaround is broken, because it allows `// STOPSHIP breaks preview`, which is not good!

**Comment by TWiStErRob** [comment]
I think we need something like
```yaml
  ForbiddenComment:
    valuePatterns: [ '^(?i)REVIEW\b', '^STOPSHIP\b' ]
```

Which would be exclusively allowed to be present in the configuration only when `values` is not present.

**Comment by schalkms** [comment]
I think the config would be more flexible if allows just one regular expression. I like your idea.
@TWiStErRob may I ask if you have resources to submit a PR and help out?

**Comment by TWiStErRob** [comment]
No time at the moment from my side, maybe later (months). If anyone else is willing, feel free to take it.

**Comment by TWiStErRob** [comment]
@schalkms how come you say one regex is more flexible than multiple? Consider a scenario where we have multiple rules, say:
- comment shouldn't start with a known todo prefix (todo, fixme, stopship)  
but if it contains a Jira ticket number, then it's ok.
- [comments should be sentences](https://nedbatchelder.com/blog/201401/comments_should_be_sentences.html) (i.e. shouldn't end with non-punctuation)
- comment shouldn't contain known procrastinating terms ("will do later", "another time", etc.)

Each of these could be complex by themselves, and combining each into one monster regex may not be the best idea.

**Comment by schalkms** [comment]
> how come you say one regex is more flexible than multiple?

I didn't write that. What I meant is that one regular expression is more flexible than the currently existing configuration option.

**Comment by TWiStErRob** [comment]
Oh, I see! 😅

**Comment by TWiStErRob** [comment]
I'm not sure about deprecation, probably a yes, but needs clear doc on common use cases (regex is hard for some).

If not deprecated probably both `values` and `valuesPatterns` shouldn't be allowed together, or clearly defined if it's OR (my vote) or AND relationship. But then the default might be problematic if someone doesn't explicitly override `values: []`, they can't really use patterns. Based on this, I would go full on for pattern.

**Comment by schalkms** [comment]
This should be deprecated or even removed and described in the *breaking changes* release notes section.
I don't think it makes any sense to maintain both options. This would create even new problems. Detekt users are usually aware on how to use regex in detekt configs. 😉

> I'm not sure about deprecation, probably a yes, but needs clear doc on common use cases (regex is hard for some).
